engine/runtime/src/main/scala/org/enso/compiler/phase/ImportResolver.scala

```diff
@@ -54,7 +54,9 @@ class ImportResolver(compiler: Compiler) {
             val importedModules = ir.imports.flatMap {
               case imp: IR.Module.Scope.Import.Module =>
                 val impName = imp.name.name
-                val exp     = ir.exports.find(_.name.name == impName)
+                val exp = ir.exports
+                  .collect { case ex: IR.Module.Scope.Export.Module => ex }
+                  .find(_.name.name == impName)
```

Comment:
Given you're touching the imports resolver, how much work would it be to fix #1569?

Reply:
More than I can spend :P Probably ~2 days.